src/systems/systems/src/adirs/mod.rs

```diff
@@ -245,6 +246,12 @@ impl AirDataInertialReferenceSystem {
             })
             .unwrap_or_else(|| Duration::from_secs(0))
     }
+
+    fn number_aligned(&self) -> f64 {
+        self.adirus
+            .iter()
+            .fold(0, |n, adiru| n + adiru.is_aligned() as i32) as f64
+    }
```

Comment:
**thought**: Can make use of `filter` and `count`.
**suggestion**: See below, note that you need to add a `Write` implementation for `usize` for this in `systems/src/simulation/mod.rs`.

```suggestion
    fn number_aligned(&self) -> usize {
        self.adirus
            .iter()
            .filter(|adiru| adiru.is_aligned()).count()
    }
```